qiskit/transpiler/__init__.py

```diff
@@ -155,15 +228,599 @@
+      - Default.  Uses `Qiskit's modified Sabre routing algorithm <sabre-lightsabre-paper_>`_ to swap
+        map.
+
+    * - :ref:`none <transpiler-preset-stage-routing-none>`
+      - Disable routing.  Raises an error if routing is required.
+
+    * - :ref:`basic <transpiler-preset-stage-routing-basic>`
+      - Greedy swap insertion to route a single operation at a time.
+
+    * - :ref:`stochastic <transpiler-preset-stage-routing-stochastic>`
+      - Consider operations layer-by-layer, using a stochastic algorithm to find swap networks that
+        implement a suitable permutation to make the layer executable.
+
+    * - :ref:`lookahead <transpiler-preset-stage-routing-lookahead>`
+      - Breadth-first search with heuristic pruning to find swaps that make gates executable.
+
+.. _transpiler-preset-stage-routing-none:
+
+Built-in ``none`` plugin
+........................
+
+A dummy plugin used to disable routing entirely.  This can occasionally be useful for
+hardware-configuration experiments, or in certain special cases of partial compilation.
+
+.. _transpiler-preset-stage-routing-basic:
+
+Built-in ``basic`` plugin
+.........................
+
+Uses the :class:`.BasisSwap` greedy swap-insertion algorithm.  This is conceptually very simple; for
+each operation in topological order, insert the shortest-path swaps needed to make the connection
+executable on the device.
+
+The optimization level only affects the amount of work the :class:`.VF2PostLayout` step does to
+attempt to improve the initial layout after routing.
+
+This method typically has poor output quality.
+
+.. _transpiler-preset-stage-routing-stochastic:
+
+Built-in ``stochastic`` plugin
+..............................
+
+Uses the :class:`.StochasticSwap` algorithm to route.  In short, this stratifies the circuit into
+layers, then uses a stochastic algorithm to find a permutation that will allow the layer to execute,
+and a series of swaps that will implement that permutation in a hardware-valid way.
+
+The optimization level affects the number of stochastic trials used for each layer, and the amount
+of work spent in :class:`.VF2PostLayout` to optimize the initial layout.
+
+This was Qiskit's primary routing algorithm for several years, until approximately 2021.  Now, it
+is reliably beaten in runtime and output quality by :ref:`Qiskit's custom Sabre-based routing
+algorithm <transpiler-preset-stage-routing-sabre>`.
+
+.. _transpiler-preset-stage-routing-lookahead:
+
+Built-in ``lookahead`` plugin
+.............................
+
+Uses the :class:`.LookaheadSwap` algorithm to route.  Approximately, this is a breadth-first search
+at producing a swap network, where the tree being explored is pruned down to a small number of
+candidate swaps at each depth.
+
+This algorithm is somewhat akin to the ``basic`` heuristic of :ref:`the "sabre" plugin
+<transpiler-preset-stage-routing-sabre>`, except it considers the following effects of each swap to
+a small depth as well.
```

Comment:
```suggestion
a small degree as well.
```

Reply:
"depth" is a technical term in the context of tree searches here